src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1361,6 +1361,8 @@ ValueTask<Stream> GetStreamInternal(FieldDescription field, int ordinal, bool as
             throw new InvalidOperationException("A stream is already open for this reader");
 
         var t = SeekToColumn(ordinal, async, cancellationToken);
+        if (_isSequential)
```

Comment:
@roji a bit late, but shouldn't we check this before calling `SeekToColumn`?

Reply:
I don't think so; if the sequential reader is currently on some previous column, and now GetStream is called on a later column, SeekToColumn gets us to the start of the later column, and so CheckColumnStart won't complain. If we're already at the requested column, SeekToColumn does nothing; CheckColumnStart will then either throw (if we're in the middle if the column), or be OK (if we're at the start; this can happen e.g. if the user did IsDBNull and then GetStream).

Makes sense?

Reply:
The problem is, `SeekToColumn` can be called asynchronously, so at that point it might (or might not) have completed.

So, depending on your luck, you can get either non-zero:

https://github.com/npgsql/npgsql/blob/4e7f2a7c9edc1a3172f552e36e26c1e6184b5377/src/Npgsql/NpgsqlDataReader.cs#L2116-L2122

Or zero:

https://github.com/npgsql/npgsql/blob/4e7f2a7c9edc1a3172f552e36e26c1e6184b5377/src/Npgsql/NpgsqlDataReader.cs#L2139-L2142

Reply:
Ahh, an `await` is missing on the SeekToColumn - is that what you mean?

Reply:
It's not missing per se: we handle it below with `GetStreamLong`.

Reply:
I think it's probably better to add the check after ` if (!t.IsCompleted)` check, and in `GetStreamLong`.

Reply:
Yeah, if we want to keep the "super-high-perf" aspect. But this seems a bit exaggerated just for avoiding a (synchronously-returning) async call for GetStreamAsync - I'd say, let's just make GetStreamInternal async and be done with it. But you're right that there's a bug here - open a separate issue?

Reply:
Agree, opened #4518.